src/main/java/org/spongepowered/api/data/type/MatterTypes.java

```diff
@@ -24,23 +24,22 @@
  */
 package org.spongepowered.api.data.type;
 
-import org.spongepowered.api.Sponge;
+import org.spongepowered.api.ResourceKey;
+import org.spongepowered.api.registry.Registries;
 
+import java.util.Locale;
 import java.util.function.Supplier;
 
-public final class MatterTypes {
+public enum MatterTypes implements Supplier<MatterType> {
 
-    // SORTFIELDS:ON
+    GAS,
+    LIQUID,
+    SOLID;
 
-    public static final Supplier<MatterType> GAS = Sponge.getRegistry().getCatalogRegistry().provideSupplier(MatterType.class, "gas");
+    private final Supplier<MatterType> supplier = Registries.MATTER_TYPE.getSupplier(ResourceKey.sponge(this.name().toLowerCase(Locale.ROOT)));
```

Comment:
This approach works here, but would not work with `SoundTypes` for example - I think it would be much better to pass the key into the constructor:
```java
GAS(ResourceKey.sponge("gas"))
```
Especially since some need special casing as well.